### PR TITLE
[FOR REVIEW ONLY] Design refresh and component re-architecture

### PR DESCRIPTION
**Pull request by davidrleonard**
I already merged the prerelease/refresh code into `master`, and there's no good way to do reviews on Github without a Pull Request, so I'm doing it like this. Don't merge this PR. We'll just use it to compare the new code vs. the code that used to be in px-context-browser.

@mdwragg @Menkhaus-ge Have at it with the code review. Any and all questions, thoughts, etc. are welcome.

### CODE REVIEW

File: sass/px-context-browser.scss [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+:host([collapsed]) px-context-browser-panel {
+  position: relative;
+  z-index: 2;
+}
+
+.column {
+  display: block;
+  position: absolute;
+  left: 0;
```
Comment (davidrleonard) [inline sass/px-context-browser.scss]: We do need `top` and `left` here.

File: sass/px-context-browser-filter.scss [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+:host {
+  display: none;
+}
+
+:host([active]) {
+  display: block;
+}
+
+.search__form {
```
Comment (davidrleonard) [inline sass/px-context-browser-filter.scss]: Fine to rename

File: sass/px-context-browser-columns.scss [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// Settings
+@import "px-defaults-design/_settings.defaults.scss";
+@import "px-colors-design/_settings.colors.scss";
+
+// Generic
+@import "px-normalize-design/_generic.normalize-mini.scss";
+@import "px-box-sizing-design/_generic.box-sizing.scss";
+
+// Component
+#columns > ::slotted(:not(.active):not(.animating)) {
```
Comment (davidrleonard) [inline sass/px-context-browser-columns.scss]: Make sure you test this after changing. Please also check the corresponding html file (px-context-browser-columns.html) and see if we use the `#columns` id in our JavaScript (`this.$.columns` or `Polymer.dom(this.root).querySelector('#columns')`). If we don't use it, please remove the id completely the HTML.

File: sass/px-context-browser-columns.scss [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * The copyright to the computer software herein is the property of
+ * GE Global Research. The software may be used and/or copied only
+ * with the written permission of GE Global Research or in accordance
+ * with the terms and conditions stipulated in the agreement/contract
+ * under which the software has been supplied.
+ */
+
+// Settings
+@import "px-defaults-design/_settings.defaults.scss";
+@import "px-colors-design/_settings.colors.scss";
```
Comment (davidrleonard) [inline sass/px-context-browser-columns.scss]: 👍

File: sass/px-context-browser-panel.scss [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+ * The copyright to the computer software herein is the property of
+ * GE Global Research. The software may be used and/or copied only
+ * with the written permission of GE Global Research or in accordance
+ * with the terms and conditions stipulated in the agreement/contract
+ * under which the software has been supplied.
+ */
+
+// Settings
+@import "px-defaults-design/_settings.defaults.scss";
+@import "px-colors-design/_settings.colors.scss";
```
Comment (davidrleonard) [inline sass/px-context-browser-panel.scss]: 👍

File: sass/px-context-browser-panel.scss [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+.pointer {
+  height: 16px;
+  width: 16px;
+  margin-left: 15px;
+  position: relative;
+  z-index: 1;
+  border-left: 16px solid transparent;
+  border-right: 16px solid transparent;
+  border-bottom: 16px solid var(--px-context-browser-panel-border-color, $gray3);
```
Comment (davidrleonard) [inline sass/px-context-browser-panel.scss]: 👍

File: sass/px-context-browser-panel.scss [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+.panel-body {
+  display: block;
+  position: relative;
+  width: $px-context-browser-panel-width;
+  border-style: solid;
+  border-width: $px-context-browser-panel-border-size;
+  border-color: $px-context-browser-panel-border-color;
+  box-shadow: $px-context-browser-panel-box-shadow;
```
Comment (davidrleonard) [inline sass/px-context-browser-panel.scss]: Please test the result.

File: sass/px-context-browser.scss [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+// Settings
+@import "px-defaults-design/_settings.defaults.scss";
+@import "px-colors-design/_settings.colors.scss";
+
+// Generic
+@import "px-normalize-design/_generic.normalize.scss";
+@import "px-box-sizing-design/_generic.box-sizing.scss";
+
+// Component
+.items {
```
Comment (davidrleonard) [inline sass/px-context-browser.scss]: 👍  just delete this

File: sass/px-context-browser-columns.scss (hunk 3 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-columns.scss]: `this.$.columns` is used in the JavaScript

File: px-context-browser-actions.html [hunk 9]
```diff
@@ -0,0 +1,94 @@
+<!-- Action: Select -->
```
Comment (mdwragg) [inline px-context-browser-actions.html]: Add `polymer.html` include.

File: px-context-browser-actions.html [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+      }
+      :host(:hover) svg,
+      :host([active]) svg {
+        stroke: var(--px-context-browser-icon-select-stroke, darkolivegreen);
+        fill: var(--px-context-browser-icon-select-fill, yellowgreen);
+      }
+    </style>
+
+    <!-- Source code copied from px-icon-set.html to allow for color styling -->
+    <svg viewBox="0 0 32 32">
```
Comment (mdwragg) [inline px-context-browser-actions.html]: Is this now needed after IconGate2017?

File: px-context-browser-actions.html [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        stroke-width: 2px;
+      }
+      :host(:hover) svg,
+      :host([active]) svg {
+        stroke: var(--px-context-browser-icon-favorite-stroke, darkgoldenrod);
+        fill: var(--px-context-browser-icon-favorite-fill, yellow);
+      }
+    </style>
+
+    <!-- Source code copied from px-icon-set.html to allow for color styling -->
```
Comment (mdwragg) [inline px-context-browser-actions.html]: See above, do we still need this after IconGate?

File: px-context-browser-columns.html [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+      <slot name="columns"></slot>
+    </div>
+  </template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'px-context-browser-columns',
+
+    properties: {
+      active: {
```
Comment (mdwragg) [inline px-context-browser-columns.html]: Add JSDoc for properties. Sensible defaults too where possible.

File: px-context-browser-filter.html [hunk 13]
```diff
@@ -0,0 +1,93 @@
+<link rel="import" href="../polymer/polymer.html"/>
+<link rel="import" href="../iron-icon/iron-icon.html"/>
+<link rel="import" href="../px-icon-set/px-icon-set.html"/>
+<link rel="import" href="css/px-context-browser-filter-styles.html"/>
+
+<dom-module id="px-context-browser-filter">
+  <template>
+    <style include="px-context-browser-filter-styles"></style>
+    <!-- <iron-a11y-keys id="a11y" target="[[_searchEl]]" keys="tab" on-keys-pressed="_handleTabKey"></iron-a11y-keys>
```
Comment (mdwragg) [inline px-context-browser-filter.html]: Please remove commented out code

File: px-context-browser-filter.html [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+          placeholder="Filter"
+          value="{{filter::input}}"
+          autocomplete="off"
+          tabindex="0"
+          on-focus="_handleFocus"
+          on-blur="_handleBlur">
+      </input>
+      <template is="dom-if" if="[[_showFilterIcon(filter)]]">
+        <iron-icon class="search__icon" icon="px-utl:filter"></iron-icon>
+        <!-- @TODO: Figure out why px-icon won't work here, it is not resizing to respect the width/height the way iron-icon does -->
```
Comment (mdwragg) [inline px-context-browser-filter.html]: Please pull this comment out for prod release - capture via GH Issues etc.

File: px-context-browser-filter.html [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+        <iron-icon id="close" class="search__icon search__icon--tappable" tabindex="-1" icon="px-nav:close" on-tap="clearFilter"></iron-icon>
+      </template>
+    </div>
+  </template>
+  <script>
+    Polymer({
+      is: 'px-context-browser-filter',
+
+      properties: {
+        filter: {
```
Comment (mdwragg) [inline px-context-browser-filter.html]: Please add JSDoc comments for properties.

File: px-context-browser-filter.html [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+        filtering: {
+          type: Boolean,
+          value: false,
+          notify: true,
+          computed: '_isFilterActive(filter)'
+        }
+      },
+
+      _handleFocus(evt) {
+        evt.stopPropagation();
```
Comment (mdwragg) [inline px-context-browser-filter.html]: Do we need event normalisation at all here? Perhaps not given just stopping propagation but worth checking 👍

File: px-context-browser-header.html [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+      </template>
+    </div>
+  </template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'px-context-browser-header',
+
+    properties: {
+      label: {
```
Comment (mdwragg) [inline px-context-browser-header.html]: Please add JSdoc comments for properties.

File: px-context-browser-item.html [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+      </template>
+    </div>
+  </template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'px-context-browser-item',
+
+    properties: {
+      label: {
```
Comment (mdwragg) [inline px-context-browser-item.html]: Please add JSdoc comments for props

File: px-context-browser-panel.html [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+      <slot name="columns"></slot>
+    </div>
+  </template>
+</dom-module>
+<script>
+  Polymer({
+    is: 'px-context-browser-panel',
+
+    properties: {
+      showLoadingBar: {
```
Comment (mdwragg) [inline px-context-browser-panel.html]: Please add JSdoc comments for props

File: px-context-browser-trigger.html [hunk 20]
```diff
@@ -0,0 +1,82 @@
+<link rel="import" href="../px-icon-set/px-icon-set.html"/>
```
Comment (mdwragg) [inline px-context-browser-trigger.html]: Please add `polymer.html` import

File: px-context-browser-trigger.html [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+      detached() {
+        this._setTrigger(null);
+      },
+
+      listeners: {
+        'tap' : '_handleTapped'
+      },
+
+      _handleTapped(evt) {
+        evt.preventDefault();
```
Comment (mdwragg) [inline px-context-browser-trigger.html]: Do we need to normalise event? Maybe not given simple usage. Just worth checking...

File: sass/px-context-browser-columns.scss (hunk 3 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-columns.scss]: fixed

File: sass/px-context-browser-filter.scss (hunk 2 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-filter.scss]: fixed

File: sass/px-context-browser-panel.scss (hunk 7 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-panel.scss]: fixed

File: sass/px-context-browser-columns.scss (hunk 4 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-columns.scss]: fixed

File: sass/px-context-browser-panel.scss (hunk 5 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-panel.scss]: fixed

File: sass/px-context-browser-panel.scss (hunk 6 above)
Comment (Menkhaus-ge) [inline sass/px-context-browser-panel.scss]: fixed

**Pull request closed without merging**